### PR TITLE
@sweir27: Add full bid increments array

### PR DESCRIPTION
**Pull request by craigspaeth**
This populates the full array of increments you would need to render the max bid select...

image

### CODE REVIEW

File: test/schema/sale_artwork.js [hunk 1]
```diff
@@ -104,19 +104,25 @@ describe('SaleArtwork type', () => {
...
+          bid_increments
         }
       }
     `;
     return graphql(schema, query)
-      .then(({ data }) => data.sale_artwork.bid_increment.should.equal(2));
+      .then(({ data }) => {
+        data.sale_artwork.bid_increments.slice(0, 20).should.eql([
+          350000, 355000, 360000, 365000, 370000, 375000, 380000, 385000,
+          390000, 395000, 400000, 410000, 420000, 430000, 440000, 450000,
```
Comment (craigspaeth) [inline test/schema/sale_artwork.js]: You can see how it jumps from increments of `5000` to `10000` after `400000` here.

File: schema/sale_artwork.js [hunk 2]
```diff
@@ -215,16 +218,27 @@ const SaleArtworkType = new GraphQLObjectType({
...
+        type: new GraphQLList(GraphQLInt),
         resolve: ({ minimum_next_bid_cents }) => {
           return gravity('/increments').then((res) => {
             const deflt = find(res, { key: 'default' });
-            const bucket = find(deflt.increments, (inc) =>
-              minimum_next_bid_cents >= inc.from &&
-              minimum_next_bid_cents <= inc.to
-            );
-            return bucket.amount;
+            const increments = [minimum_next_bid_cents];
```
Comment (craigspaeth) [inline schema/sale_artwork.js]: This got rather hairy to implement—I might be missing a more novel solution. Basically this mutates and array 100 times keeping track of the last bid to predict if the next one is going to land in a new bucket.

Comment (acjay) [inline schema/sale_artwork.js]: Ooo, a puzzler. How about:

```
function increments(from, to) {
  function loop(amount, [tier, ...restTiers]) {
    let nextAmount, nextTiers, nextRestTiers;
    if (amount + tier.amount > tier.to) {
      if (restTiers.length == 0) return [];
      [nextTier, ...nextRestTiers] = restTiers;
      nextAmount = amount + newTier.amount;
    } else {
      nextTier = tier;
      restTiers = restTiers;
      nextAmount = amount + tier.amount;
    }
    if (nextAmount > to) return [];
    return [nextAmount, ...loop(nextAmount, nextTier, ...nextRestTiers)];
  }
  return loop(from, deflt.increments);
}
```

No idea if that works!

I think the original idea was to make the top of the range some multiple of the max of the current price and the high estimate. But an arbitrary top end should suffice for now!

Comment (acjay) [inline schema/sale_artwork.js]: Or, embrace mutability and do a generator:

```
function* increments(from, to) {
  let tiers = deflt.increments.slice();
  for (let next = from; next <= to; next += tiers[0].amount) {
    if (next > tiers[0].to && tiers.length > 0) tiers = tiers.slice(1);
    yield next;
  }
}
```

Then you can do `[...increments(1000, 500000)]`

**Comment by acjay** [comment]
@ashfurrow brought up the point that this is a lot of data to send, when all the lots in the sale share an increment policy. As seen in https://github.com/artsy/eigen/issues/1633#issuecomment-226260771 and elsewhere, we're already at the point where the data sent over to initialize the clients is causing noticeable delays to start up the sale. I think we're going to have to find ways to be more data-frugal. Seems like we probably want to just send over the raw policy structure, and let the clients calculate the increments, even though that logic will be redundant.

**Comment by craigspaeth** [comment]
@acjay @ashfurrow That's a fair point. We can ship over the raw increment policy as a root query in MP too—is that something I should try to wedge in quickly or is it too late at this point? It's also [available in Gravity](https://api.artsy.net/api/v1/increments) if that helps.

**Comment by ashfurrow** [comment]
Wedging it in would be awesome, if you could Craig.

**Comment by craigspaeth** [comment]
@ashfurrow PR sent https://github.com/artsy/metaphysics/pull/319

**Comment by acjay** [comment]
Thanks Craig!

**Pull request merged**